collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2176,59 +2175,50 @@ public void denied() {
 
-                    if (!showFirst) {
-                        // we've just loaded a saved form, so start in the hierarchy view
-                        String formMode = reqIntent.getStringExtra(ApplicationConstants.BundleKeys.FORM_MODE);
-                        if (formMode == null || ApplicationConstants.FormModes.EDIT_SAVED.equalsIgnoreCase(formMode)) {
-                            formControllerAvailable(formController);
-                            identityPromptViewModel.requiresIdentityToContinue().observe(this, requiresIdentity -> {
-                                if (!requiresIdentity) {
-                                    if (!allowMovingBackwards) {
-                                        // we aren't allowed to jump around the form so attempt to
-                                        // go directly to the question we were on last time the
-                                        // form was saved.
-                                        // TODO: revisit the fallback. If for some reason the index
-                                        // wasn't saved, we can now jump around which doesn't seem right.
-                                        FormIndex formIndex = SaveFormIndexTask.loadFormIndexFromFile();
-                                        if (formIndex != null) {
-                                            formController.jumpToIndex(formIndex);
-                                            onScreenRefresh();
-                                            return;
-                                        }
+                    // we've just loaded a saved form, so start in the hierarchy view
+                    String formMode = reqIntent.getStringExtra(ApplicationConstants.BundleKeys.FORM_MODE);
+                    if (formMode == null || ApplicationConstants.FormModes.EDIT_SAVED.equalsIgnoreCase(formMode)) {
+                        identityPromptViewModel.formLoaded(formController);
+                        identityPromptViewModel.requiresIdentityToContinue().observe(this, requiresIdentity -> {
+                            if (!requiresIdentity) {
+                                if (!allowMovingBackwards) {
+                                    // we aren't allowed to jump around the form so attempt to
+                                    // go directly to the question we were on last time the
+                                    // form was saved.
+                                    // TODO: revisit the fallback. If for some reason the index
+                                    // wasn't saved, we can now jump around which doesn't seem right.
+                                    FormIndex formIndex = SaveFormIndexTask.loadFormIndexFromFile();
+                                    if (formIndex != null) {
+                                        formController.jumpToIndex(formIndex);
+                                        formControllerAvailable(formController);
```

Comment:
Replacing `onScreenRefresh` with `formControllerAvailable` is an untested change. We don't currently have a good way to test save point flows - I'll look at backfilling that for work on #5240. It might be that we want to remove this particular flow (returning to the exact index on a save point load when moving backwards is disabled). Otherwise, it looks like we might want to investigate if there is missing audit logging here.

Reply:
But you were able to verify this change manually at least right?

Reply:
Yes. Sorry I should have said that. @getodk/testers to be very specific about the untested flow here: disable moving backwards and then reload a form from a save point (kill the app during form entry and then restart).